content/blog/2016/2016-08-10-rails-cd-with-pipeline.adoc

```diff
@@ -0,0 +1,358 @@
+---
+layout: post
+title: "Continuous Security for Rails apps with Pipeline and Brakeman"
+tags:
+- ruby
+- pipeline
+- rails
+- brakeman
+- continuousdelivery
+author: rtyler
+---
+
+:sectanchors:
+
+
+[NOTE]
+--
+This is a guest post by link:https;//github.com/rtyler[R. Tyler Croy], who is a
+long-time contributor to Jenkins and the primary contact for Jenkins project
+infrastructure. He is also a Jenkins Evangelist at
+link:http://cloudbees.com[CloudBees, Inc.]
+--
+
+
+When the link:http://rubyonrails.org[Ruby on Rails] framework debuted it
+changed the industry in two noteworthy ways: it created a trend of opinionated web
+application frameworks (link:https://www.djangoproject.com/[Django],
+link:https://playframework.com/[Play], link:https://grails.org/[Grails]) and it
+also _strongly_ encouraged thousands of developers to embrace test-driven
+development along with many other modern best practices (source control, dependency
+management, etc). Because Ruby, the language underneath Rails, is interpreted
+instead of compiled there isn't a "build" per se but rather tens, if not
+hundreds, of tests, linters and scans which are run to ensure the application's
+quality. With the rise in popularity of Rails, the popularity of application
+hosting services with easy-to-use deployment tools like link:http://heroku.com[Heroku] or
+link:http://engineyard[Engine Yard] rose too.
+
+This combination of good *test coverage* and easily *automated deployments*
+makes Rails easy to continuously deliver with Jenkins. In this post we'll cover
+testing non-trivial Rails applications with link:/doc/pipeline[Jenkins
+Pipeline] and, as an added bonus, we will add security scanning via
+link:http://brakemanscanner.org[Brakeman] and the
+link:https://wiki.jenkins-ci.org/display/JENKINS/Brakeman+Plugin[Brakeman
+plugin].
+
+image::/images/post-images/ruby-pipeline-2016/cfpapp-stage-view.png[role=center]
+
+*Topics*
+
+* <<Preparing the app>>
+* <<Preparing Jenkins>>
+* <<Writing the Pipeline>>
+* <<Security scanning>>
+* <<Deploying the good stuff>>
+* <<Wrap up>>
+
+
+For this demonstration, I used link:http://rubycentral.org/[Ruby Central]'s
+link:https://github.com/rubycentral/cfp-app[cfp-app], a:
+
+[quote]
+____
+Ruby on Rails application that lets you manage your conference's call for
+proposal (CFP), program and schedule. It was written by Ruby Central to run the
+CFPs for RailsConf and RubyConf.
+____
+
+
+This isn't just some random Rails app I'm using for a pretty demo, this is
```

Comment:
And so what if it WAS just some random Rails app?! You could just strike that part and start with, "This is the application we used..." I don't feel strongly about it though, your choice.